docs/pages/api-docs/data-grid/grid-api.md

```diff
@@ -88,8 +88,8 @@ import { GridApi } from '@material-ui/x-grid';
 | <span class="prop-name">state</span>                                                              | <span class="prop-type">GridState</span>                                                                                                                                                 | Property that contains the whole state of the grid.                                                                                                            |
 | <span class="prop-name">subscribeEvent</span>                                                     | <span class="prop-type">(event: string, handler: (params: any, event?: SyntheticEvent&lt;Element, Event&gt;) =&gt; void, options?: GridSubscribeEventOptions) =&gt; () =&gt; void</span> | Registers a handler for an event.                                                                                                                              |
 | <span class="prop-name">toggleColumnMenu</span>                                                   | <span class="prop-type">(field: string) =&gt; void</span>                                                                                                                                | Toggles the column menu under the `field` column.                                                                                                              |
-| <span class="prop-name">updateColumn</span>                                                       | <span class="prop-type">(col: GridColDef) =&gt; void</span>                                                                                                                              | Updates the definition of a column.                                                                                                                            |
-| <span class="prop-name">updateColumns</span>                                                      | <span class="prop-type">(cols: GridColDef[]) =&gt; void</span>                                                                                                                           | Updates the definition of multiple columns at the same time.                                                                                                   |
+| <span class="prop-name">updateColumn</span>                                                       | <span class="prop-type">(col: GridColDef, shouldApplyFlex?: boolean) =&gt; void</span>                                                                                                   | Updates the definition of a column.                                                                                                                            |
```

Comment:
It's not great to add this argument here

Reply:
It's still a draft :+1: 
I was redacting the description of the PR.
I have cases with `shouldApplyFlex = true` and `shouldApplyFlex = false` in `apiRef.current.updateColumn(s)` in other hooks so I can't just rely on a default value.

I can create new methods in `GridAPI` but for me it make sense when the user update the columns through the API to let him decide if he want to reapply the flex.

Reply:
I would personally expect to have `shouldApplyFlex` always true

Reply:
@oliviertassinari that means that for manual resize through the header, you could cause a shrink on the flex column down to 50px. This is a change of behavior that would encourage the users to put a width or minWidth on their flex column. Whereas today increasing the width of a column by resizing it will never cause the other columns to shrink.

But I suppose it make sense.

Reply:
@oliviertassinari 

In the doc we have the following line :

> flex doesn't work together with width. If you set both flex and width in GridColDef, flex will override width.

And indeed our current flex implementation does not care about the `width` prop
How do we want to make it evolve ?

Currently my PRs behavior is to never allow a flex column to have less computedWidth than its with

For instance :

```js
columns = [{ field: 'id', flex: 1, width: 100 }, { field: 'brand', width: 100 }]
viewportWidth = 300

=>

columnsAfterFlex = [{ field: 'id', computedWidth: 200 }, { field: 'brand', width: 100 }]
```

but 

```js
columns = [{ field: 'id', flex: 1, width: 300 }, { field: 'brand', width: 100 }]
viewportWidth = 300

=>

columnsAfterFlex = [{ field: 'id', computedWidth: 300 }, { field: 'brand', width: 100 }]
```

Whereas today we would have 

```js
columns = [{ field: 'id', flex: 1, width: 300 }, { field: 'brand', width: 100 }]
viewportWidth = 300

=>

columnsAfterFlex = [{ field: 'id', computedWidth: 200 }, { field: 'brand', width: 100 }]
```

But if we want to keep ignoring the `width` property when `flex` is defined. Then the user should do the following in order to avoid having its flex column squashed to much.

```js
columns = [{ field: 'id', flex: 1, minWidth: 300 }, { field: 'brand', width: 100 }]
viewportWidth = 300

=>

columnsAfterFlex = [{ field: 'id', computedWidth: 300 }, { field: 'brand', width: 100 }]
```

The two can make sens. I find the 1st one a bit more straightforward.